src/ucp/api/ucp.h

```diff
@@ -1015,6 +1027,30 @@ typedef struct ucp_params {
 } ucp_params_t;
 
 
+/**
+ * @ingroup UCP_CONTEXT
+ * @brief Lib attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the Library.
+ */
+typedef struct ucp_lib_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_lib_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t              field_mask;
+
+    /**
+     * Thread support of the library. For supported thread levels please
+     * see @ref ucs_thread_mode_t.
+     */
+    ucs_thread_mode_t     thread_mode;
```

Comment:
either "max_thread_level"/"max_thread_mode"
or uin64_t thread_modes /* bitmap of supported thread modes */